clippy_lints/src/ok_if_let.rs

```diff
@@ -0,0 +1,53 @@
+use rustc::lint::*;
+use rustc::hir::*;
+use utils::{paths, method_chain_args, span_help_and_lint, match_type};
+
+/// **What it does:*** Checks for unnecessary `ok()` in if let.
+///
+/// **Why is this bad?** Calling `ok()` in if let is unnecessary, instead match on `Ok(x`
+///
+/// **Known problems:** None.
+///
+/// **Example:**
+/// ```rustc
+/// for result in iter {
+///     if let Some(bench) = try!(result).parse().ok() {
+///         vec.push(bench)
+///     }
+/// }
+/// ```
+declare_lint! {
+    pub IF_LET_SOME_RESULT,
+    Warn,
+    "usage of `ok()` in `if let Some(x)` statements is unnecessary, match on `Ok(expr)` instead"
```

Comment:
For consistency this should be `` `Some(x) … `Ok(x)` `` or `` `Some(pat)` … `Ok(pat)` ``.
(`Ok(expr)` is wrong, this is pattern matching)